src/proxy/inbound.rs

```diff
@@ -134,12 +135,29 @@ impl Inbound {
                 match futures_util::future::select(Box::pin(drain.signaled()), serve).await {
                     // We got a shutdown request. Start gracful shutdown and wait for the pending requests to complete.
                     futures_util::future::Either::Left((_shutdown, mut server)) => {
+                        debug!("inbound serve got drain {:?}", server);
                         let drain = std::pin::Pin::new(&mut server);
                         drain.graceful_shutdown();
-                        server.await
+                        // There are scenarios where the http2 server never resolves after
+                        // `graceful_shutdown`, which will hang the whole task.
```

Comment:
Interesting. So before `inpod`, drain was the entire process shutting down. Now its not, so it MUST terminate in some bounded time.

Do we want this here, or at the `inpod` layer?

Reply:
> Do we want this here, or at the inpod layer?

I'm not sure where else to put it, but in terms of error reporting this seems best - this task is spawned to run to completion, and the inpod layer already initiates the drain and waits for it, the most it can do is drop the drain.

(we could do spawnhandles and aborts and punch this up a layer I guess, but the effect would be the same)